src/Microsoft.HttpRepl.Tests/FileSystem/RealFileSystemTests.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.HttpRepl.FileSystem;
+using Xunit;
+
+namespace Microsoft.HttpRepl.Tests.FileSystem
+{
+    public class RealFileSystemTests
```

Comment:
Add tests for the validation failure cases?
- null
- extension doesn't start with "."
- extension is only "." (assuming that's invalid... well, you could test it either way)

Reply:
Added more tests for those scenarios.